src/qengine/qengine.cpp

```diff
@@ -145,6 +145,11 @@ bitCapInt QEngine::ForceM(const bitLenInt* bits, const bitLenInt& length, const
 
 void QEngine::ApplySingleBit(const complex* mtrx, bool doCalcNorm, bitLenInt qubit)
 {
+    if (doNormalize && !isUnitary2x2(mtrx)) {
+        throw("Nonunitary gates and normalization cannot be used at the same time. (You probably shouldn't use "
```

Comment:
Remove the opinion statements :)

Reply:
(Don't use nonunitary gates! They don't even _exist!_ It's not an aesthetic issue!) I'll word it in a way that makes the reason for the standard clear, rather.